### PR TITLE
doc: monovertex

### LINKED ISSUE #2090: Document MonoVertex

### PR DESCRIPTION
**Pull request by vigith**
Resolves #2090

### CODE REVIEW

File: docs/core-concepts/monovertex.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# MonoVertex
+
+MonoVertex is a simplified version of the [Pipeline](./pipeline.md). The major design idea behind MonoVertex is to simplify data-processing
+in those cases where independently autoscaling vertices are not required. This means there will only be a single vertex 
+called the MonoVertex, which is capable of running the [Source](../user-guide/sources/overview.md), the [Sink](../user-guide/sinks/overview.md)
+(with [Fallback Sink](../user-guide/sinks/fallback.md) if need be), and optionally the [Transformer](../user-guide/sources/transformer/overview.md) 
+together. There is no concept of Edges in MonoVertex since there is only one [Vertex](vertex.md). The MonoVertex runs the same containers run by the 
+[Pipeline](./pipeline.md), this means the user’s can switch between MonoVertex and [Pipeline](./pipeline.md) by just
```
Comment (whynowy) [inline docs/core-concepts/monovertex.md:8]:
```suggestion
[Pipeline](./pipeline.md), this means the users can switch between MonoVertex and [Pipeline](./pipeline.md) by just 
```

File: docs/core-concepts/monovertex.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# MonoVertex
+
+MonoVertex is a simplified version of the [Pipeline](./pipeline.md). The major design idea behind MonoVertex is to simplify data-processing
+in those cases where independently autoscaling vertices are not required. This means there will only be a single vertex 
+called the MonoVertex, which is capable of running the [Source](../user-guide/sources/overview.md), the [Sink](../user-guide/sinks/overview.md)
+(with [Fallback Sink](../user-guide/sinks/fallback.md) if need be), and optionally the [Transformer](../user-guide/sources/transformer/overview.md) 
+together. There is no concept of Edges in MonoVertex since there is only one [Vertex](vertex.md). The MonoVertex runs the same containers run by the 
+[Pipeline](./pipeline.md), this means the user’s can switch between MonoVertex and [Pipeline](./pipeline.md) by just 
+changing the spec. MonoVertex’s [autoscaling](../specifications/autoscaling.md)is also similar to the [Pipeline](./pipeline.md),
```
Comment (whynowy) [inline docs/core-concepts/monovertex.md:9]:
```suggestion
changing the spec. MonoVertex’s [autoscaling](../specifications/autoscaling.md)is also similar to the Source vertex of a [Pipeline](./pipeline.md), 
```

File: docs/core-concepts/monovertex.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+* Very high TPS can be supported because there are no ISBs (RAFT consensus) in between.
+* Extremely low latency because there is no network hops between the Vertices (only one Vertex).
+* Low Operational overhead, no need to maintain ISBs.
+
+_MonoVertex - Condensed Pipeline with just one Vertex_
+![monovertex.png](../assets/monovertex.png)
+
+
+## Use cases of MonoVertex
+There are a few scenarios where MonoVertex is the best fit, which can be best described as use-cases where “the user is
```
Comment (whynowy) [inline docs/core-concepts/monovertex.md:23]: a blank line is needed.

File: docs/core-concepts/monovertex.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+reading from the Source, doing simple transformations, and writing to a Sink”. You don't need to scale these Source,
+Sink, or Map vertices by themselves. If the Sink can't keep up, you just need to add more pods. This brings a new set
+of Source, Transformer, and a Sink (optionally with a backup sink). The same goes for if the Source cannot keep up
+(pending is increasing), we just need to add more pods.
+
+### A Brief History
+
+We developed MonoVertex because at Intuit we found that, there are a decent amount of applications that read from
```
Comment (whynowy) [inline docs/core-concepts/monovertex.md:31]: Not to use "first person" and "Intuit" here?

**Review by KeranYang — APPROVED** [review]

**Pull request merged**